### ISSUE TITLE
Destroying ibm_compute_vm_instance completes before instance is destroyed

### ISSUE THREAD

**Issue by czernitko** [IBM-Cloud/terraform-provider-ibm#855]
Hi there,
we use destroy-time Terraform provisioner to perform a cleanup after removal of  `ibm_compute_vm_instance` resource. The action performed in the destroy-time provisioner can be performed only after the instance is actually removed from the account (or at least shut down). 
Current behaviour is, that the destroy-time provisioner is executed within a few seconds after running the destroy from Terraform. Instance is marked as being deleted in the UI, but it is still running and it takes some time before it actually shuts down and is removed.

### Terraform Version
0.11.14

### Affected Resource(s)
- ibm_compute_vm_instance

### Terraform Configuration Files
```hcl
resource "ibm_compute_vm_instance" "vm_instance" {
  count = "${var.count}"

  hostname          = "${var.vm_hostname}"
  domain            = "${var.domain}"
  os_reference_code = "${var.os_reference_code}"
  ...
  provisioner "local-exec" {
    when        = "destroy"
    command     = "... doing the cleanup ..."
    interpreter = ["bash", "-c"]
  }
}
```

### Expected Behavior
The bash command in destroy-time provisioner should be executed on the machine running Terraform only after the IBM instance is actually removed from the account (or at least completely shut down). Running it after the instance is actually removed would be preferable for folks who are rebuilding the nodes and want to re-use some resources I guess.

### Actual Behavior
The destroy time provisioner is called once the instance removal command is accepted, ie. within a few seconds, not waiting for the instance to be removed. This might be ok for CLI behaviour not to block the console, but doesn’t play well with Terraform. If the resource is removed by Terraform, it is expected not to exist anymore.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Define `ibm_compute_vm_instance` with destroy-time provisioner and count > 1.
2. Run `terraform apply` to create the instance(s)
3. Remove the instance from IBM by removing it from Terraform configuration or scaling it down (count = 0)
4. Check the IBM Cloud Console - the instance still exists, although the destroy-time provisioner was already executed.

### References
- Similar issue: https://github.com/IBM-Cloud/terraform-provider-ibm/issues/271

**Comment by r2ronoha** [comment]
I have the same issue.
On destroy, `ibm_compute_vm_instance` reports as `Destruction complete` before the instance is actually removed, which creates issues when doing clean up that requires the instance to be deleted.

**Comment by hkantare** [comment]
closing the issue due to no recent activity. Please feel free to open new issue
